Modules/_gdbmmodule.c

```diff
@@ -654,20 +654,43 @@ static struct PyModuleDef _gdbmmodule = {
 
 PyMODINIT_FUNC
 PyInit__gdbm(void) {
-    PyObject *m, *d, *s;
+    PyObject *m;
 
     if (PyType_Ready(&Dbmtype) < 0)
             return NULL;
     m = PyModule_Create(&_gdbmmodule);
-    if (m == NULL)
+    if (m == NULL) {
         return NULL;
-    d = PyModule_GetDict(m);
+    }
+
     DbmError = PyErr_NewException("_gdbm.error", PyExc_OSError, NULL);
-    if (DbmError != NULL) {
-        PyDict_SetItemString(d, "error", DbmError);
-        s = PyUnicode_FromString(dbmmodule_open_flags);
-        PyDict_SetItemString(d, "open_flags", s);
-        Py_DECREF(s);
+    if (DbmError == NULL) {
+        goto error;
+    }
+    Py_INCREF(DbmError);
```

Comment:
Do we need this? `PyErr_NewException` returns a new reference.

Reply:
I'm not sure, but it seems like most modules do the same:

socktemodule.c:

```
    socket_herror = PyErr_NewException("socket.herror",
                                       PyExc_OSError, NULL);
    if (socket_herror == NULL)
        return NULL;
    Py_INCREF(socket_herror);
    PyModule_AddObject(m, "herror", socket_herror);
```

signalmodule.c:

```
    ItimerError = PyErr_NewException("signal.ItimerError",
            PyExc_OSError, NULL);
    if (ItimerError != NULL)
        PyDict_SetItemString(d, "ItimerError", ItimerError);
```

Note: PyModule_AddObject() is weird, it decrements the reference counter on success...

Reply:
It's documented to steal a reference, though from the implementation, only on success. Module init functions seem usually not written carefully about this kind error. @serhiy-storchaka

Reply:
> Module init functions seem usually not written carefully about this kind error.

If you want to fix that, I would suggest to fix it in a different PR. Maybe even open a new bug to track that.